prow/cmd/deck/job_history.go

```diff
@@ -240,16 +286,45 @@ func (bucket gcsBucket) listBuildIDs(root string) ([]int64, error) {
 	return ids, nil
 }
 
-func parseJobHistURL(url *url.URL) (bucketName, root string, buildID int64, err error) {
+// parseJobHistURL parses the job History URL
+// example urls:
+// * new format: https://prow.k8s.io/job-history/s3/kubernetes-jenkins/pr-logs/directory/pull-capi?buildId=1245584383100850177
+// * old format: https://prow.k8s.io/job-history/kubernetes-jenkins/pr-logs/directory/pull-capi?buildId=1245584383100850177
+// Newly generated URLs will include the storageType. We still support old URLs so they don't break.
+// For old URLs we assume that the storageType is `gs`.
+// examples return values:
+// * storageType: s3, gs
+// * bucketName: kubernetes-jenkins
+// * root: pr-logs/directory/pull-capi
+// * buildID: 1245584383100850177
+func parseJobHistURL(url *url.URL) (storageType, bucketName, root string, buildID int64, err error) {
 	buildID = emptyID
 	p := strings.TrimPrefix(url.Path, "/job-history/")
-	s := strings.SplitN(p, "/", 2)
-	if len(s) < 2 {
-		err = fmt.Errorf("invalid path (expected /job-history/<gcs-path>): %v", url.Path)
-		return
+	// examples for p:
+	// * new format: s3/prow-artifacts/pr-logs/directory/bazel-build-test
+	// * old format: kubernetes-jenkins/pr-logs/directory/pull-cluster-api-provider-openstack-test
+	if strings.HasPrefix(p, "gs") || strings.HasPrefix(p, "s3") {
```

Comment:
please change that to `gs/`/`s3/` so it doesn't match a bucketName that starts with `gs` or `s3`

Reply:
Also, lets simplify this and not have two completely different handlings:

* If there is no provider in the url path, inject `gs`
* Only handle three element urls and remove the old handling (The error needs to mention both options though as we wont know which one was used by the time we emit it)

Reply:
Fixed